### ISSUE TITLE
Git-upstream fails when doing a `keptn update project` and changing files in the repo

### ISSUE THREAD

**Issue by johannes-b** [keptn/keptn#2125]
The following procedure results in a bug: 

1. Create a project (without Git upstream)
1. Create a Git repo
1. Update the project by setting the git-remote-url and git credentials: `keptn update project PROJECTNAME --git-user=GIT_USER --git-token=GIT_TOKEN --git-remote-url=GIT_REMOTE_URL`
1. Change a file in the repo (e.g. on GitHub) 
1. Send a new-artifact event does not work since git commit does not work

**Logs of the configuration-service:** 
```
{"timestamp":"2020-07-27T08:59:42.774391415Z","logLevel":"DEBUG","message":"Staging Changes"}
{"timestamp":"2020-07-27T08:59:43.31659198Z","logLevel":"ERROR","message":"Could not commit to staging branch of project keptn07project"}
{"timestamp":"2020-07-27T08:59:43.316619254Z","logLevel":"ERROR","message":"Error executing command git push https://grabnerandi:********@github.com/grabnerandi/keptn07__simplenode.git: exit status 1\nTo https://github.com/grabnerandi/keptn07__simplenode.git\n ! [rejected]        staging -\u003e staging (fetch first)\nerror: failed to push some refs to 'https://grabnerandi:********@github.com/grabnerandi/keptn07__simplenode.git'\nhint: Updates were rejected because the remote contains work that you do\nhint: not have locally. This is usually caused by another repository pushing\nhint: to the same ref. You may want to first integrate the remote changes\nhint: (e.g., 'git pull ...') before pushing again.\nhint: See the 'Note about fast-forwards' in 'git push --help' for details.\n"}
``` 

**Problem:**
* The branches in the local repository are not set to the branches on the upstream repo.

**Fix:**
* Set remote branch using: `git branch --set-upstream-to=origin/dev dev` or `git push -u origin HEAD`

**Comment by grabnerandi** [comment]
The solution to this is explained in the following stackoverflow post: https://stackoverflow.com/questions/32056324/there-is-no-tracking-information-for-the-current-branch

Here is the important piece:
![image](https://user-images.githubusercontent.com/675886/88537929-bef3f500-d00e-11ea-8ea1-b1a247936366.png)

This means that when doing a "keptn update project" the configuration-service additionally needs to iterate through all branches and setup branch trecking with:
```
git branch --set-upstream-to=origin/BRANCHNAME BRANCHNAME
``` 

I validated this manually and it fixed my installation

**Comment by grabnerandi** [comment]
@checkelmann was also able to reproduce this behavior in a local test

**Comment by checkelmann** [comment]
I will prepare a fix for this issue. 
Need to extend the Addorigin function to loop over all bracnhes, check them out and add the remote to them.

**Comment by agrimmer** [comment]
Hi @checkelmann,
I already implemented the fix in #2129
